src/models/release.ts

```diff
@@ -729,6 +729,153 @@ const getReleaseModel = function (
 		remove: tagsModel.remove,
 	};
 
+	/**
+	 * @namespace asset
+	 * @memberof balena.models.release
+	 */
+	const asset = {
+		/**
+		 * @summary Get all release assets for a release
+		 * @name getAllByRelease
+		 * @public
+		 * @function
+		 * @memberof balena.models.release.asset
+		 *
+		 * @param {String|Number|Object} commitOrIdOrRawVersion - release commit, ID or raw version object
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object[]} - release assets
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.release.asset.getAllByRelease(123).then(function(assets) {
+		 * 	console.log(assets);
+		 * });
+		 *
+		 * @example
+		 * balena.models.release.asset.getAllByRelease('7cf02a6').then(function(assets) {
+		 * 	console.log(assets);
+		 * });
+		 */
+		getAllByRelease: assetsModel.getAllByRelease,
+
+		/**
+		 * @summary Get a specific release asset
+		 * @name get
+		 * @public
+		 * @function
+		 * @memberof balena.models.release.asset
+		 *
+		 * @param {Number|Object} id - release asset ID or unique identifier object
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object} - release asset
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.release.asset.get(123).then(function(asset) {
+		 * 	console.log(asset);
+		 * });
+		 *
+		 * @example
+		 * balena.models.release.asset.get({
+		 * 	asset_key: 'logo.png',
+		 * 	release: 123
+		 * }).then(function(asset) {
+		 * 	console.log(asset);
+		 * });
+		 */
+		get: assetsModel.get,
+
+		/**
+		 * @summary Download a release asset
+		 * @name download
+		 * @public
+		 * @function
+		 * @memberof balena.models.release.asset
+		 *
+		 * @param {Number|Object} id - release asset ID or unique identifier object
+		 * @fulfil {NodeJS.ReadableStream} - download stream
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.release.asset.download(123).then(function(stream) {
+		 * 	stream.pipe(fs.createWriteStream('logo.png'));
+		 * });
+		 *
+		 * @example
+		 * balena.models.release.asset.download({
+		 * 	asset_key: 'logo.png',
+		 * 	release: 123
+		 * }).then(function(stream) {
+		 * 	stream.pipe(fs.createWriteStream('logo.png'));
+		 * });
+		 */
+		download: assetsModel.download,
+
+		/**
+		 * @summary Upload a release asset
+		 * @name upload
+		 * @public
+		 * @function
+		 * @memberof balena.models.release.asset
+		 *
+		 * @param {Object} uploadParams - upload parameters
+		 * @param {String|File} uploadParams.asset - asset file path (Node.js) or File object (browser)
```

Comment:
Isn't the File approach working on both browsers & node?
Maybe b/c of that benefit, File should be the primarily advertised way of using this method
```suggestion
		 * @param {String|File} uploadParams.asset - asset file path (Node.js) or File object (Node.js & browser)
```

Reply:
Yes :) good catch